apollo-federation/src/operation/optimize.rs

```diff
@@ -847,7 +847,9 @@ impl Selection {
                     // Expand the fragment
                     let expanded_sub_selections =
                         fragment.selection_set.retain_fragments(fragments_to_keep)?;
-                    if *parent_type == fragment.spread.data().type_condition_position {
+                    if *parent_type == fragment.spread.data().type_condition_position.clone()
```

Comment:
```suggestion
                    if *parent_type == fragment.spread.data().type_condition_position
```

This clone is weird. We are only using the value inline immediately so it can't be necessary to clone if we are using the right reference type.